pkg/volume/cinder/cinder.go

```diff
@@ -227,6 +228,34 @@ func (plugin *cinderPlugin) ConstructVolumeSpec(volumeName, mountPath string) (*
 	return volume.NewSpecFromVolume(cinderVolume), nil
 }
 
+var _ volume.ExpandableVolumePlugin = &cinderPlugin{}
+
+func (plugin *cinderPlugin) ExpandVolumeDevice(spec *volume.Spec, newSize resource.Quantity, oldSize resource.Quantity) (resource.Quantity, error) {
+	cinder, _, err := getVolumeSource(spec)
+	if err != nil {
+		glog.Errorf("error when getting volume source : %v", err)
```

Comment:
more log-and-returns all the way through here.  I won't bother calling them out individually.